### ISSUE TITLE
letsencrypt failure - openhabian proxy setup script

### ISSUE THREAD

**Issue by robshep** [openhab/openhabian#539]
```
Creating Let's Encrypt certificate...
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Plugins selected: Authenticator webroot, Installer None
Obtaining a new certificate
Performing the following challenges:
http-01 challenge for <me>.duckdns.org
Using the webroot path /var/www/<me>.duckdns.org for all unmatched domains.
Waiting for verification...
Cleaning up challenges
Failed authorization procedure. <me>.duckdns.org (http-01): urn:ietf:params:acme:error:unauthorized :: The client lacks sufficient authorization :: Invalid response from http://<me>.duckdns.org/.well-known/acme-challenge/XfRbx26Qka4VGXghbLR8XtbLG3O_ujNFU2T7x2ciTcE [81.151.148.159]: "<html>\r\n<head><title>401 Authorization Required</title></head>\r\n<body bgcolor=\"white\">\r\n<center><h1>401 Authorization Required</"

IMPORTANT NOTES:
 - The following errors were reported by the server:

   Domain: <me>.duckdns.org
   Type:   unauthorized
   Detail: Invalid response from
   http://<me>.duckdns.org/.well-known/acme-challenge/XfRbx26Qka4VGXghbLR8XtbLG3O_ujNFU2T7x2ciTcE
   [81.151.148.159]: "<html>\r\n<head><title>401 Authorization
   Required</title></head>\r\n<body
   bgcolor=\"white\">\r\n<center><h1>401 Authorization Required</"

   To fix these errors, please make sure that your domain name was
   entered correctly and the DNS A/AAAA record(s) for that domain
   contain(s) the right IP address.
```

**Comment by robshep** [comment]
Looks to me like some bits of nginx are commented out - but maybe they are removed outside of the letsencrypt process


```
$ cat /etc/nginx/sites-enabled/openhab
## Let's Encrypt webroot location
#   location /.well-known/acme-challenge/ {
#       root                                    /var/www/<me>.duckdns.org;
#   }
```

**Comment by BClark09** [comment]
Hi @robshep, looks like the NGINX setup routine is broken, I've submitted a fix for it #540 but I have yet to test it, would you be able to do so?

**Comment by robshep** [comment]
Sure thing.

Can I ask the best method for testing?

Can I just clone the PR branch into a file tree? Or is it a built version that is installed?

Or do I cherry pick a few files etc?


Cheers

Rob

Get Outlook for iOS<https://aka.ms/o0ukef>

________________________________

**Comment by altair2010** [comment]
i can try to test the patch, because i'm blocked by the bug.
But, the PR #540 has conflicts

**Comment by altair2010** [comment]
@robshep to test the PR on your config

```
cd /opt/openhabian
sudo wget https://github.com/openhab/openhabian/pull/540.patch
sudo git am 540.patch
```
source : https://coderwall.com/p/6aw72a/creating-patch-from-github-pull-request

**Comment by mstormi** [comment]
Fixed by #818
